web/ui/module/codemirror-promql/src/parser/parser.test.ts

```diff
@@ -825,6 +825,124 @@ describe('promql operations', () => {
       expectedValueType: ValueType.vector,
       expectedDiag: [],
     },
+    {
+      expr: '{"foo"}',
+      expectedValueType: ValueType.vector,
+      expectedDiag: [],
+    },
+    {
+      // with metric name in the middle
+      expr: '{a="b","foo",c~="d"}',
+      expectedValueType: ValueType.vector,
+      expectedDiag: [],
+    },
+    {
+      expr: '{"foo", a="bc"}',
+      expectedValueType: ValueType.vector,
+      expectedDiag: [],
+    },
+    {
+      expr: '{"colon:in:the:middle"}',
+      expectedValueType: ValueType.vector,
+      expectedDiag: [],
+    },
+    {
+      expr: '{"dot.in.the.middle"}',
+      expectedValueType: ValueType.vector,
+      expectedDiag: [],
+    },
+    {
+      expr: '{"😀 in metric name"}',
+      expectedValueType: ValueType.vector,
+      expectedDiag: [],
+    },
+    {
+      // quotes with escape
+      expr: '{"this is \"foo\" metric"}', // eslint-disable-line
+      expectedValueType: ValueType.vector,
+      expectedDiag: [],
+    },
+    {
+      expr: '{"foo","colon:in:the:middle"="val"}',
+      expectedValueType: ValueType.vector,
+      expectedDiag: [],
+    },
+    {
+      expr: '{"foo","dot.in.the.middle"="val"}',
+      expectedValueType: ValueType.vector,
+      expectedDiag: [],
+    },
+    {
+      expr: '{"foo","😀 in label name"="val"}',
```

Comment:
can you also add an error case for invalid UTF-8 characters? In Go, that's done with this function: https://cs.opensource.google/go/go/+/refs/tags/go1.22.0:src/unicode/utf8/utf8.go;l=575

here's the equivalent go test: https://github.com/prometheus/common/blob/main/expfmt/text_parse_test.go#L563-L567

Reply:
Sure! I'll add the case for invalid UTF-8 characters

Reply:
Just to clarify,  `metric_name{label="\xbd"}` cannot be allowed because `\xbd` is not a valid utf-8 string right?
And this follows for metric name and label names right?

Reply:
Precisely.

Reply:
I did some research on this but I couldn't find any methods to check this like Golang's `utf8.ValidString()`. I can only think of using regex for validation but I think it's quite overwhelming and error prone.

Reply:
the implementation of ValidString in go is not too too complex, and you may be able to reimplement it in typescript: https://cs.opensource.google/go/go/+/refs/tags/go1.22.0:src/unicode/utf8/utf8.go;l=528

Reply:
Yeah I think the same can be implemented in typescript also. I'll try !

Reply:
If it gets too involved to do that, I think it is fine to not validate for valid UTF-8 sequences in the frontend code. The backend will still catch it if it happens, and it is extremely unlikely to enter an invalid UTF-8 character by accident (as you cannot really _type_ it directly).

(The validation in the backend is important because programs could, accidentally or maliciously, put random bit sequences into a label, and we sometimes rely on the strings to be valid UTF-8 internally, for example by using separator characters that are not valid UTF-8.)

Reply:
with what @beorn7 said, I think we don't even need to implement this validation in typescript.

Reply:
Alright then. I think I've implemented all the requested changes.

Also it was getting hard to implement the `validUTF8String` as JS uses UTF-16 encoding.